src/Bicep.Cli/Services/Validation.cs

```diff
@@ -1,33 +1,49 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.IO;
 using System.Text;
 using Bicep.Core.Emit;
 using Bicep.Core.Intermediate;
 using Bicep.Core.Semantics;
 using Bicep.Core.Syntax;
+using Microsoft.Extensions.Logging;
 using Microsoft.WindowsAzure.ResourceStack.Common.Collections;
 using Microsoft.WindowsAzure.ResourceStack.Common.Json;
 using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
 
 namespace Bicep.Cli.Services
 {
-    public class ValidationService
+    public class Validation
     {
-        public static void Validate(ImmutableArray<TestSymbol> testDeclarations)
+        public InsensitiveDictionary<Evaluation> SuccessfullEvaluations { get; }
+
+        public InsensitiveDictionary<Evaluation> FailedEvaluations { get; }
+
+        public InsensitiveDictionary<Evaluation> SkippedEvaluations { get; }
+
+        public bool Success => FailedEvaluations.Count == 0 && SkippedEvaluations.Count == 0;
+
+        public int TotalEvaluations => SuccessfullEvaluations.Count + FailedEvaluations.Count + SkippedEvaluations.Count;
+
+        public Validation(ImmutableArray<TestSymbol> testDeclarations)
+        {
+            SuccessfullEvaluations = new InsensitiveDictionary<Evaluation>();
+            FailedEvaluations = new InsensitiveDictionary<Evaluation>();
+            SkippedEvaluations = new InsensitiveDictionary<Evaluation>();
+
+            Validate(testDeclarations);
+        }
+        public void Validate(ImmutableArray<TestSymbol> testDeclarations)
```

Comment:
To address @shenglol's point about immutability, what about making the validation class static, and having it return results. Given the rename I suggested above, that would mean you could call it as follows:

```csharp
TestResults results = TestEvaluator.Run(IEnumerable<TestSymbol> tests);
```